irods/connection.py

```diff
@@ -65,31 +65,37 @@ def __init__(self, pool, account):
         self._disconnected = False
 
         scheme = self.account._original_authentication_scheme
-        auth_type = ''
+
+        # These variables are just useful diagnostics.  The login_XYZ() methods should fail by
+        # raising exceptions if they encounter authentication errors.
+        auth_module = auth_type = ''
 
         if self.server_version >= (4,3,0):
+            auth_module = None
             # use client side "plugin" module: irods.auth.<scheme>
             irods.auth.load_plugins(subset=[scheme])
             auth_module = getattr(irods.auth, scheme, None)
+
+            # TODO (#518 unrelated): gsi module that also sets _client_ctx?
```

Comment:
Capture TODOs as issues and remove from the code.

TODOs in code give off the impression that something is incomplete. They're also difficult to track.

Reply:
We don't support GSI authentication in 4.3+, so I'm not sure whether this needs to be considered. And it's not just that we would fall back to legacy GSI authentication in the case that scheme is used, but we actually literally do not build the plugin in 4.3 and so the server will return an error if it is attempted because there is no plugin to load for that scheme.

---

That being said, I now have a counter to/rant about the general TODO comment. :p I ultimately yield, of course, but I'm not sure that I agree with this.

> TODOs in code give off the impression that something is incomplete.

... could be stated conversely as...

> Missing TODOs in code give off the impression that nothing is incomplete.

In this case, something is in fact incomplete, and so isn't it even even less likely that the incomplete-ness is rectified when there is nothing to indicate that something is incomplete?

Secondly, I don't think it's difficult to track these because there's an issue number next to it and we can/will link this line in that issue so that attention is brought to the TODO. I guess it says "unrelated" here, but I'm just speaking more generally.

Just my two cents.

Reply:
My comment about the TODOs could have been worded a little differently. What I meant was that they can give off that impression and they can be lost as the code changes.

I'm open to changing how we handle TODOs if you or others on the team feel it's beneficial.

Reply:
I see, the idea being that they might be removed without actually accomplishing what they are flagging as incomplete? I personally don't see that as enough of a drawback to exclude them for the potential benefit that they give. I also don't feel strongly enough about it to think we really need a process around this since we have many ways of tracking TODO items. This is just one more indication that something still needs doing.

Again, just my two cents!

Reply:
With the resolution of irods/irods_docs#252, I think this can be resolved?

Reply:
Actually, we may still be able to remove the TODO given the comment I made about GSI. Let's discuss...

Reply:
@d-w-moore Given that 4.3 doesn't include support for GSI auth, what does that TODO comment mean?

Reply:
> @d-w-moore Given that 4.3 doesn't include support for GSI auth, what does that TODO comment mean?

Ah, that does change things then.  Will delete the TODO. Anyway, I noticed [this line](https://github.com/irods/python-irodsclient/blob/ed2e73cb79dcf08efedeed4a19f3e2a43db90f9c/irods/connection.py#L84) and the setting of client_ctx was the reason for the TODO ;  if we were continuing to support, I was going to introduce a gsi module that at least made sure that member was set.

Reply:
We can add it if we ever have need of a GSI authentication plugin, but I don't see why we need to mention it. To me, even mentioning it here suggests that it exists or has ever existed, which is does not and has not. Thoughts?

Reply:
Agreed. Let's avoid mentioning GSI since it's no longer supported.

Reply:
torch it

Reply:
done